app/jobs/bulkrax/create_relationships_job.rb

```diff
@@ -21,63 +21,60 @@ class CreateRelationshipsJob < ApplicationJob
 
     queue_as :import
 
-    attr_accessor :base_entry, :child_record, :parent_record, :importer_run
+    attr_accessor :child_records, :parent_record, :parent_entry, :importer_run_id
 
-    # @param entry_identifier [String] source_identifier of the base Bulkrax::Entry the job was triggered from (see #build_for_importer)
-    # @param parent_identifier [String] Work/Collection ID or Bulkrax::Entry source_identifier
-    # @param child_identifier [String] Work/Collection ID or Bulkrax::Entry source_identifier
+    # @param parent_identifier [String] Work/Collection ID or Bulkrax::Entry source_identifiers
     # @param importer_run [Bulkrax::ImporterRun] current importer run (needed to properly update counters)
     #
     # The entry_identifier is used to lookup the @base_entry for the job (a.k.a. the entry the job was called from).
     # The @base_entry defines the context of the relationship (e.g. "this entry (@base_entry) should have a parent").
     # Whether the @base_entry is the parent or the child in the relationship is determined by the presence of a
     # parent_identifier or child_identifier param. For example, if a parent_identifier is passed, we know @base_entry
     # is the child in the relationship, and vice versa if a child_identifier is passed.
-    def perform(entry_identifier:, parent_identifier: nil, child_identifier: nil, importer_run:)
-      @base_entry = Entry.find_by(identifier: entry_identifier)
-      @importer_run = importer_run
-      if parent_identifier.present?
-        @child_record = find_record(entry_identifier)
-        @parent_record = find_record(parent_identifier)
-      elsif child_identifier.present?
-        @parent_record = find_record(entry_identifier)
-        @child_record = find_record(child_identifier)
-      else
-        raise ::StandardError, %("#{entry_identifier}" needs either a child or a parent to create a relationship)
+    def perform(parent_identifier:, importer_run_id:)
+      pending_relationships = Bulkrax::PendingRelationship.where(bulkrax_importer_run_id: importer_run_id,
+                                                                 parent_id: parent_identifier).sort_by(&:order)
+      @importer_run_id = importer_run_id
+      @parent_record = find_record(parent_identifier)
+      @child_records = { works: [], collections: [] }
+      pending_relationships.each do |rel|
```

Comment:
this should be find_each so it doesn't load them all in memory first (find_each does pages of 500 to prevent memory bloat)